Expose client tools auto update for find endpoint

vapopov opened the pull request:
In this PR added exposing client auto update fields (is enabled, and required client tools version for update)

Related: https://github.com/gravitational/teleport/pull/39805/files#diff-8d3c0408dd826f27e9f13cd15f09f4d7e89202862f0b13ce131a2e3072c2a40dR49-R53

vapopov committed:
Expose client tools auto update for find endpoint

rosstimothy reviewed (commented):
Can you add test coverage?

vapopov commented on api/client/webclient/webclient.go:
```diff
@@ -297,6 +297,10 @@ type PingResponse struct {
 	ServerVersion string `json:"server_version"`
 	// MinClientVersion is the minimum client version required by the server.
 	MinClientVersion string `json:"min_client_version"`
+	// ToolsVersion defines the version of {tsh, tctl} for client auto update.
+	ToolsVersion string `json:"tools_version"`
+	// ToolsAutoupdate enables client auto update feature.
+	ToolsAutoupdate bool `json:"tools_autoupdate"`
```
grouping fields to another struct would be better, will change this one

vapopov committed:
Group auto update settings in find response
Log error instead returning error
Add tests auto update settings in find endpoint
Add check for not implemented error

vapopov commented on lib/web/apiserver.go:
```diff
@@ -1526,7 +1526,23 @@ func (h *Handler) find(w http.ResponseWriter, r *http.Request, p httprouter.Para
+
+	autoUpdateVersion, err := h.GetAccessPoint().GetAutoUpdateVersion(r.Context())
+	if err != nil && !trace.IsNotFound(err) {
+		return nil, trace.Wrap(err)
```
I added logging instead, but we might generate a lot of log messages with this error message in frequent requests to this endpoint

vapopov committed:
Add more test cases

sclevine commented on api/client/webclient/webclient.go:
```diff
@@ -328,6 +330,14 @@ type ProxySettings struct {
+	// ToolsVersion defines the version of {tsh, tctl} for client auto update.
+	ToolsVersion string `json:"tools_version"`
+	// ToolsAutoupdate enables client auto update feature.
+	ToolsAutoupdate bool `json:"tools_autoupdate"`
```
We should decide if we want to treat auto-update as a single hyphenated word (`toolsAutoupdate`/`tools_autoupdate`) or an abbreviated way of writing "automatic update" (`toolAutoUpdate`/`tools_auto_update`), and then be consistent about this.

Seems like @rosstimothy is suggesting two words, so maybe we move all of the JSON/YAML to the two-word form, and reserve `autoupdate` for the package, resource, and subcommand naming (`autoupdate_config`, `autoupdate` package, `tctl autoupdate client`?
